src/tool_operate.c

```diff
@@ -464,20 +463,25 @@ static CURLcode retrycheck(struct OperationConfig *config,
         }
       }
     }
+    if(!sleeptime && !config->retry_delay_ms) {
+      if(!per->retry_sleep)
+        per->retry_sleep = RETRY_SLEEP_DEFAULT;
+      else
+        per->retry_sleep *= 2;
+      if(per->retry_sleep > RETRY_SLEEP_MAX)
+        per->retry_sleep = RETRY_SLEEP_MAX;
+    }
+    if(!sleeptime)
+      sleeptime = per->retry_sleep;
     warnf("Problem %s. "
-          "Will retry in %ld second%s. "
+          "Will retry in %ld.%03ld second%s. "
```

Comment:
I was playing around with this and it looks like it's possible in a single format to show the decimal only if there's milliseconds, like this
~~~
  long sleeptime = 3000;
  printf("Will retry in %ld%s%.*ld seconds. ",
    sleeptime/1000L,
    (sleeptime%1000L ? "." : ""),
    (sleeptime%1000L ? 3 : 0),
    sleeptime%1000L);
~~~